### PR TITLE
Improve error message when NodeJS is not installed

### PR DESCRIPTION
**Pull request by hanwen-pcluste**
API: NodeJS is checked at the start of flask app to ensure the API can run all the pcluster commands.
CLI: Only check for NodeJS when necessary, i.e. when running `create-cluster`, `update-cluster`, `build-image`.

Signed-off-by: Hanwen <hanwenli@amazon.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/src/pcluster/api/util.py [hunk 1]
```diff
@@ -148,3 +149,16 @@ def _deserialize_dict(data, boxed_type):
...
     """
     return {k: _deserialize(v, boxed_type) for k, v in six.iteritems(data)}
+
+
+def assert_node_executable(logger):
+    node_exe = shutil.which("node")
+    logger.debug("Found nodejs executable in %s", node_exe)
+    if not node_exe:
+        message = (
+            "You must install Node.js in order to use ParallelCluster. Please install Node.js from https://nodejs.org"
```
Comment (enrico-usai) [inline cli/src/pcluster/api/util.py:159]: Adding the specific version of node in the message requires us to change the code every now and then to keep this help update so let's use a more generic message or better point to CDK documentation.

I'd say: 
```
Node.js is required to use ParallelCluster. 
It is a prerequisite of AWS CDK library used by ParallelCluster, 
see installation instructions here: https://docs.aws.amazon.com/cdk/latest/guide/work-with.html#work-with-prerequisites
```

Comment (enrico-usai) [reply cli/src/pcluster/api/util.py:159]: @nstornetta what do you suggest?

File: README.md [hunk 2]
```diff
@@ -34,13 +34,8 @@ Make sure you have installed the [AWS Command Line Interface](https://docs.aws.a
...
 
-Install Node Version Manager and Node.js
-```
-wget -qO- https://raw.githubusercontent.com/nvm-sh/nvm/v0.38.0/install.sh | bash
-. ~/.nvm/nvm.sh
-nvm install node
-node --version
-```
+[Node.js](https://nodejs.org/en/) is required by AWS CDK library used by ParallelCluster.
+You could see installation instructions [in AWS CDK documentaton](https://docs.aws.amazon.com/parallelcluster/latest/ug/install-v3.html).
```
Comment (charlesg3) [inline README.md:38]: ~You could see~ :arrow_right: Please reference 

in **the** AWS CDK documentation

File: cli/src/pcluster/api/controllers/cluster_operations_controller.py [hunk 3]
```diff
@@ -96,6 +97,7 @@ def create_cluster(
     (Defaults to &#39;true&#39;.)
     :type rollback_on_failure: bool
     """
+    assert_node_executable(LOGGER)
```
Comment (charlesg3) [inline cli/src/pcluster/api/controllers/cluster_operations_controller.py:100]: Is it possible to just add the `LOGGER` to the file that has the `assert_node_executable` function?

File: cli/tests/pcluster/cli/test_build_image.py [hunk 4]
```diff
@@ -112,6 +112,26 @@ def test_error(self, mocker, test_datadir):
             run(command)
         assert_that(exc_info.value.data).is_equal_to(api_response[0])
 
+    def test_no_nodejs_error(self, mocker, test_datadir):
+        """Test expected message is printed out if nodejs is not installed."""
+        mocker.patch(
```
Comment (charlesg3) [inline cli/tests/pcluster/cli/test_build_image.py:117]: Looks like this fits on a single line (?)

File: cli/tests/pcluster/cli/test_create_cluster.py [hunk 5]
```diff
@@ -208,6 +208,26 @@ def test_error(self, mocker, test_datadir):
             run(command)
         assert_that(exc_info.value.data).is_equal_to(api_response[0])
 
+    def test_no_nodejs_error(self, mocker, test_datadir):
+        """Test expected message is printed out if nodejs is not installed."""
+        mocker.patch(
```
Comment (charlesg3) [inline cli/tests/pcluster/cli/test_create_cluster.py:213]: Single line here too (?)

**Review by charlesg3 — COMMENTED** [review]
A few comments inlined.

File: README.md (hunk 2 above)
Comment (hanwen-pcluste) [reply README.md]: Done

File: cli/src/pcluster/api/controllers/cluster_operations_controller.py (hunk 3 above)
Comment (hanwen-pcluste) [reply cli/src/pcluster/api/controllers/cluster_operations_controller.py]: Done

File: cli/tests/pcluster/cli/test_build_image.py (hunk 4 above)
Comment (hanwen-pcluste) [reply cli/tests/pcluster/cli/test_build_image.py]: Done

File: cli/tests/pcluster/cli/test_create_cluster.py (hunk 5 above)
Comment (hanwen-pcluste) [reply cli/tests/pcluster/cli/test_create_cluster.py]: Done

File: cli/src/pcluster/api/util.py (hunk 1 above)
Comment (hanwen-pcluste) [reply cli/src/pcluster/api/util.py]: Changed according latest discussion in the team

**Review by charlesg3 — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -34,13 +34,8 @@ Make sure you have installed the [AWS Command Line Interface](https://docs.aws.a
...
 
-Install Node Version Manager and Node.js
-```
-wget -qO- https://raw.githubusercontent.com/nvm-sh/nvm/v0.38.0/install.sh | bash
-. ~/.nvm/nvm.sh
-nvm install node
-node --version
-```
+[Node.js](https://nodejs.org/en/) is required by AWS CDK library used by ParallelCluster.
+Please reference installation instructions [in the AWS CDK documentaton](https://docs.aws.amazon.com/parallelcluster/latest/ug/install-v3.html).
```
Comment (enrico-usai) [inline README.md:38]: [in the AWS CDK documentaton] --> we're pointing to our documentation not AWS CDK and there is a typo in the word "documentaton"

**Pull request merged**